src/Partials/AlignBoxButton.vala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Giacomo "giacomoalbe" Alberini <giacomoalbe@gmail.com>
+ */
+public class Akira.Partials.AlignBoxButton : Gtk.Grid {
+    public string icon_name;
+    public string action;
+
+    public Gtk.IconSize icon_size;
+
+    private Gtk.Button button;
+    private Gtk.Image image;
+
+    public signal void triggered (Akira.Partials.AlignBoxButton emitter);
```

Comment:
nit, but move the signal at the very top of the declared items :) 

the order should be:

- Consts
- Signals
- public fields
- private fields 
- constructors
- Object functions